src/keyDerivation.c

```diff
@@ -0,0 +1,75 @@
+#include <os_io_seproxyhal.h>
+#include <stdint.h>
+
+#include "assert.h"
+#include "errors.h"
+#include "keyDerivation.h"
+
+#define VALIDATE_PARAM(cond) if (!(cond)) THROW(ERR_INVALID_REQUEST_PARAMETERS)
+
+void validatePath(uint32_t* path, uint32_t pathLength)
```

Comment:
I wonder where should be this validation logic. Logic between getExtendedPublicKey and deriveAddress would be slightly different...

Reply:
Let's see when we implement getExtendedPublicKey

Reply:
ok